cmd/content-sources/main.go

```diff
@@ -96,3 +109,36 @@ func apiServer(ctx context.Context, wg *sync.WaitGroup, allRoutes bool) {
 		}
 	}()
 }
+
+func instrumentation(ctx context.Context, wg *sync.WaitGroup, reg *prometheus.Registry) {
+	wg.Add(1)
+	e := echo.New()
+
+	e.Add(http.MethodGet, "/metrics", echo.WrapHandler(promhttp.HandlerFor(
+		reg,
+		promhttp.HandlerOpts{
+			// Opt into OpenMetrics to support exemplars.
+			EnableOpenMetrics: true,
+			// Pass custom registry
+			Registry: reg,
+		},
+	)))
+	e.HideBanner = true
+	go func() {
+		defer wg.Done()
+		log.Logger.Info().Msgf("Starting instrumentation")
+		if err := e.Start(":9000"); err != nil && err != http.ErrServerClosed {
```

Comment:
Just curious, why start the metrics endpoint on another port? I assumed it would be the same as our API server, but just a new endpoint. 

I couldn't find documentation anywhere to suggest one way or the other, so I was wondering why you chose this way.

Reply:
thanks for this question;  I am not sure if I have used something that I read from another repository (very likely); but now your concern is my concern :) and it made me dig about this and I have found a response that remove my uncertainty about this.

In a practical way, I have found the information about the port and the endpoint into the clowdenvirontment resource (I have reserved a namespace in the ephemeral environment and I have printed out the environment configuration by `oc get clowdenvironments.cloud.redhat.com/env-$( oc project -q ) -o yaml | yq --yaml-output '.spec.providers.metrics'`). The above provide the port (`9000`) and the path (`/metrics`) that will be used for the instrumentation.

The above information is injected by clowder, so I have to read it from the configuration and set up the endpoint according to it.

Thanks a million, I will update that!

Reply:
I've looked through some other apps, and I see some examples where metrics is an endpoint on the same server:
- [image builder](https://github.com/osbuild/image-builder/blob/d8d15007839a1a555d1fe0cd1d9ec1495d913140/internal/v1/server.go)
- [enititlements-api-go](https://github.com/RedHatInsights/entitlements-api-go/blob/c7c29cc8c8b00c8a8a1dbf1e7d16dc4584448e34/server/routes.go)

I see what you mean, but I wonder if what clowder injects is configurable from somewhere. I think if we might want to avoid it for simplicity.